client/src/game/ui/tools/DiceTool.vue

```diff
@@ -70,157 +112,584 @@ function addSelector(selector: (typeof DxConfig.selectorOptions)[number]): void
+        );
+
+        // These lines are required to make sure that the transition for lastroll-results plays,
+        // This covers the case of multiple 3D dice rolls awaiting results simultaneously. When one
+        // finishes, the awaitingRoll value is updated to false and the value is shown. If another
+        // of the rolls finishes, it will not play a transition because awaitingRoll did not change.
+        // Therefore, we change this value here to force an update and play the transition.
+        awaitingRoll.value = true;
+        await nextTick();
+
+        return result;
+    };
+
+    lastRoll.value = await getResult();
+
+    awaitingRoll.value = false;
 }
 </script>
 
 <template>
     <div id="dice" class="tool-detail">
-        <template v-if="!showHistory">
-            <div class="header">\\ SETTINGS \\</div>
-            <div id="dice-settings" class="dice-grid">
-                <label>Use 3D dice</label>
-                <ToggleGroup v-model="use3d" :options="arrToToggleGroup(_3dOptions)" :multi-select="false" />
-                <label>Share result with</label>
+        <Transition name="dice-expand">
+            <div v-show="dice3dSetting === 'box'" class="dice-roller-box">
+                <div class="dice-canvas-container">
+                    <canvas id="dice-canvas" ref="canvasElement" />
+                </div>
+            </div>
+        </Transition>
+        <div id="lastroll-container" class="history-breakdown">
+            <div id="lastroll-result">
+                <Transition :name="awaitingRoll ? 'fast-fade' : 'get-result'" mode="out-in">
+                    <div v-if="awaitingRoll">
+                        -
+                    </div>
+                    <div v-else>
+                        {{ lastRoll.result }}
+                    </div>
+                </Transition>
+            </div>
+            <div class="breakdown-scroll-container">
+                <Transition name="fast-fade" mode="out-in">
+                    <div v-if="awaitingRoll" class="breakdown-flex-container">
+                        <div style="font-style: italic">{{ t('game.ui.tools.DiceTool.rolling') }}</div>
+                    </div>
+                    <div v-else class="breakdown-flex-container">
+                        <div
+                            v-for="[index, part] of lastRoll.parts.entries()"
+                            :key="index"
+                        >
+                            <div v-if="lastRoll.result === '-'" style="font-style: italic"> <!-- starting state -->
+                                {{ part.shortResult }}
+                            </div>
+                            <div v-else-if="part.longResult" class="dice-result">
+                                <div class="input">{{ part.input ?? "" }}</div>
+                                <div class="ops">{{ '(' + part.longResult.replaceAll(',', ' + ') + ')' }}</div>
```

Comment:
I think it would be nice to add an extra option "Output" with options "short" and "long" or "expanded" or something. Which would toggle the visibility of this element.

Some people care about seeing the individual results, others just want the results and don't like the clutter of all those individual items. Having a simple toggle that can be toggled if you do need the details can be interesting I think. (And at all times have a hover message when you hover over a result show the details as well)

We can make the dice UI element a bit wider to fit 1 third toggle in that options row.

It might also be interesting to make this a markdown aware field so that the various notations for rolls that were kept or dropped also work. (See `systems/chat/md.ts` for an example, we only need a very basic markdown instance without special setups)

Reply:
> I think it would be nice to add an extra option "Output" with options "short" and "long" or "expanded" or something. Which would toggle the visibility of this element.

I agree that giving the option is a nice idea. I really don't want to increase the width of the tool if I can avoid it. Right now the width is set to nicely fit the literal+symbol clickgroup, which looks good. I also don't want to increase height because it can get pretty tall with mutiple drawers open already.

I think some sort of options menu might be nice for this. Keep things that will only be changed infrequently hidden most of the time to cut down on clutter. In the this menu, have separate options for last roll breakdown details and history item breakdown details. This also makes the tool more expandable in the future if more options are added. Obviously keep 3D dice and Share on the main screen since they're going to be used quite often.

> It might also be interesting to make this a markdown aware field so that the various notations for rolls that were kept or dropped also work.

Somehow I didn't even realize the notation was supposed to be markdown lol. I could get that working.